src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/get_file_request.cc

```diff
@@ -0,0 +1,309 @@
+      SendHMIRequest(hmi_apis::FunctionID::BasicCommunication_GetFileFromHMI,
+                     &(*message_)[strings::msg_params],
+                     true);
+      return;
+    }
+  } else {
+    LOG4CXX_ERROR(logger_, "Could not get file path");
+    SendResponse(false,
+                 mobile_apis::Result::INVALID_DATA,
+                 "Could not get file path",
+                 &response_params);
+  }
+
+  const std::string full_path = file_path + "/" + file_name_;
+  if (!file_system::FileExists(full_path)) {
+    LOG4CXX_ERROR(logger_, "File " << full_path << " does not exist");
+    SendResponse(false,
+                 mobile_apis::Result::INVALID_DATA,
+                 "File does not exist",
+                 &response_params);
+    return;
+  }
+
+  // Handle offset
+  LOG4CXX_DEBUG(logger_, "Handle offset parameter");
+  const uint64_t file_size = file_system::FileSize(full_path);
+  if (offset_ > file_size) {
+    LOG4CXX_ERROR(logger_,
+                  "Offset " << offset_ << " greater than file size "
+                            << file_size);
+    SendResponse(false,
+                 mobile_apis::Result::INVALID_DATA,
+                 "Offset greater than file size",
+                 &response_params);
+    return;
+  }
+
+  // Load data from file as binary data
+  LOG4CXX_DEBUG(logger_, "Load binary data from file");
+  std::vector<uint8_t> bin_data;
+  if (!file_system::ReadBinaryFile(full_path, bin_data, offset_)) {
+    LOG4CXX_ERROR(logger_, "Failed to read from file: " << full_path);
+    SendResponse(false,
+                 mobile_apis::Result::GENERIC_ERROR,
+                 "Unable to read from file",
+                 &response_params);
+    return;
+  }
+
+  // Construct response message
+  if ((*message_)[strings::msg_params].keyExists(strings::offset)) {
+    response_params[strings::offset] = offset_;
+  }
+  if ((*message_)[strings::msg_params].keyExists(strings::length)) {
+    response_params[strings::length] = length_;
+  }
+  if ((*message_)[strings::msg_params].keyExists(strings::file_type)) {
+    response_params[strings::file_type] = file_type_;
+  }
+  const uint32_t crc_calculated = GetCrc32CheckSum(bin_data);
+  response_params[strings::crc32_check_sum] = crc_calculated;
+
+  LOG4CXX_DEBUG(logger_, "file_name: " << file_name_);
+  LOG4CXX_DEBUG(logger_, "file_type: " << file_type_);
+  LOG4CXX_DEBUG(logger_, "offset: " << offset_);
+  LOG4CXX_DEBUG(logger_, "length: " << length_);
+  LOG4CXX_DEBUG(logger_, "file_path: " << file_path);
+  LOG4CXX_DEBUG(logger_, "crc: " << crc_calculated);
+
+  SendResponse(true,
+               mobile_apis::Result::SUCCESS,
+               "File uploaded",
+               &response_params,
+               bin_data);
+}
+
+void GetFileRequest::on_event(const app_mngr::event_engine::Event& event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  if (hmi_apis::FunctionID::BasicCommunication_GetFileFromHMI != event.id()) {
+    return;
+  }
```

Comment:
Should pass message through if you don't receive a successful result code. An example of how to do this: https://github.com/smartdevicelink/sdl_core/blob/feature/base_app_services_implementation/src/components/application_manager/rpc_plugins/app_service_rpc_plugin/src/commands/mobile/get_app_service_data_request.cc#L109